servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancer.java

```diff
@@ -508,7 +508,9 @@ private void sequentialUpdateUsedHosts(List<PrioritizedHostImpl<ResolvedAddress,
         for (PrioritizedHostImpl<?, ?> host : nextHosts) {
             host.loadBalancingWeight(host.serviceDiscoveryWeight());
         }
-        this.hostSelector = hostSelector.rebuildWithHosts(priorityStrategy.prioritize(usedHosts));
+        nextHosts = priorityStrategy.prioritize(nextHosts);
+        this.hostSelector = hostSelector.rebuildWithHosts(nextHosts);
+        loadBalancerObserver.onHostSetChanged(nextHosts);
```

Comment:
This opens a hook from an observer impl to internal LB state. Consider making the collection unmodifiable before you pass it to the observer to prevent any modifications.